### PR TITLE
feat(elasticsearch): expand captured requests

### LINKED ISSUE #916: Capture request body for more Elasticsearch APIs
See https://github.com/elastic/apm/issues/55

### PR DESCRIPTION
**Pull request by Qard**
I omitted `_rollup_search` as there appears to be no interface for it in the Node.js client. 🤔 

Fixes #916

### Checklist

- [x] Implement code
- [x] Add tests

### CODE REVIEW

**Comment by watson** [comment]
> I omitted `_rollup_search` as there appears to be no interface for it in the Node.js client. 🤔

I assume that's going to be available only in the new ES client, which we btw should add support for as well. I'll make an issue for that.

File: test/instrumentation/modules/elasticsearch.js [hunk 1]
```diff
@@ -59,6 +59,92 @@ test('client.search with callback', function userLandCode (t) {
...
+      }
+    },
+    params: {
+      q: 'pants'
+    }
+  }
+
+  resetAgent(done(t, 'POST', '/_search/template', JSON.stringify(body)))
+
+  agent.startTransaction('foo3')
```
Comment (watson) [inline test/instrumentation/modules/elasticsearch.js]: I honestly can't remember why I added the number suffix to the transaction names, but in all the previous tests I incremented it by one in each test. So I think either we should remove the suffix completely or we should continue the pattern in the new tests 🤷‍♂️

File: lib/instrumentation/modules/elasticsearch.js [hunk 2]
```diff
@@ -19,17 +19,24 @@ module.exports = function (elasticsearch, agent, { enabled }) {
...
         span.name = `Elasticsearch: ${method} ${path}`
 
-        if (query && searchRegexp.test(path)) {
-          span.setDbContext({
-            type: 'elasticsearch',
-            statement: JSON.stringify(query)
-          })
+        if (searchRegexp.test(path)) {
+          let statement = Array.isArray(body)
+            ? body.map(JSON.stringify).join('\n')
```
[the remark this answers is not in the document]
Comment (Qard) [reply lib/instrumentation/modules/elasticsearch.js]: I did it this way to copy what the actual request body looks like. If we just want the body as the JSON array, I can just stringify the body directly.

File: lib/instrumentation/modules/elasticsearch.js [hunk 3]
```diff
@@ -19,17 +19,24 @@ module.exports = function (elasticsearch, agent, { enabled }) {
       var method = params && params.method
       var path = params && params.path
       var query = params && params.query
+      var body = params && params.body
```
[the remark this answers is not in the document]
Comment (Qard) [reply lib/instrumentation/modules/elasticsearch.js]: `_count` and `_search` uses `query`, everything else uses `body`.

File: test/instrumentation/modules/elasticsearch.js (hunk 1 above)
Comment (Qard) [reply test/instrumentation/modules/elasticsearch.js]: Ah, right. 😅

Comment (Qard) [reply test/instrumentation/modules/elasticsearch.js]: fixed

File: lib/instrumentation/modules/elasticsearch.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (watson) [reply lib/instrumentation/modules/elasticsearch.js]: Ah, so it's sending it as NDJSON?

File: lib/instrumentation/modules/elasticsearch.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (watson) [reply lib/instrumentation/modules/elasticsearch.js]: Ah, didn't know that. Thanks

File: lib/instrumentation/modules/elasticsearch.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Qard) [reply lib/instrumentation/modules/elasticsearch.js]: Yes, it's ndjson. It alternates between an object describing which index to target and then the object that would normally the request body for a singular query to that index.

**Pull request merged**